Automattic/wp-calypso
# With-theme: Redirect Woo and Premium themes to the thank you page

Point the Premium and WooCommerce theme to the marketplace thank you page instead of the Design Picker.

This will replace the previous behavior introduced in this https://github.com/Automattic/wp-calypso/pull/76040 because we have themes that are present in the Showcase but not in the Design picker. 

Related to https://github.com/Automattic/dotcom-forge/issues/4355

## Proposed Changes

* Redirect the Woo and Premium themes to the marketplace page instead of pointing them to the DP.
* Fix a blank page issue in plugin-bundle on the first load.

## Testing Instructions


**Test an unlisted premium theme**
* Use the Calypso live link and go to `/start/with-theme?ref=calypshowcase&theme=smithland&theme_type=premium&style_variation=japanese-laurel&premium=true`
* Create a new site
* Notice that you are now redirected to the marketplace thank you page
* Make sure that Smithland is installed on your site
* Make sure that `japanese-laurel` style variation is used

**Test a WooCommerce theme**
* Use the live link and go to `/start/with-theme?ref=calypshowcase&theme=zaino&theme_type=woocommerce&premium=true`
* Create a new site with a Business plan
* Notice that you are redirected to Marketplace thank you page and then immediately redirected to plugin-bundle
* Make sure that `plugin-bundle` loads and you are presented with the first dedicated store
* Configure the store info.
* You should be redirected back to the Marketplace thank you page

## Pre-merge Checklist
